functions/venue.js

```diff
@@ -434,6 +434,13 @@ const removeAdmin = async (adminId) => {
     });
 };
 
+exports.createUser = functions.auth.user().onCreate(async (user) => {
+  const flag = functions.config().flag;
+  if (flag && flag.autoadmin) {
+    await addAdmin(user.uid);
+  }
+});
```

Comment:
We probably don't want to be doing this in `functions/venue.js` as it's not really related to venues. It's more of an 'auth' type thing, so should be in that file.

The name of the function doesn't make sense, we're not creating the user, so shouldn't call it `createUser`. Maybe something like `autoMakeUserAdmin` or similar.

It's probably worth noting this in a comment near this function:

> A Cloud Functions event is not triggered when a user signs in for the first time using a custom token.

You've made changes to the `functions.config()`, yet not included any of those details within this PR. There's a `*.example` file in `functions/` that will need to be updated as well.

Most of the other backend code here seems to read the functions.config() as a `const` outside of the handler itself. Any particular reason you chose to do it this way?

Reply:
> Any particular reason you chose to do it this way?

in case of a missing setting `flag.autoadmin`, should not be necessary once I update `*.example`

Reply:
And to preempt one of the likely next questions here, `addAdmin` and `removeAdmin` should be refactored into a new location. Possibly `functions/src/api/roles`. Then imported back into `functions/venue.js` for the things that need it there, and then also used from `functions/auth.js` or wherever this trigger function ends up when moved as per my comments above.

Reply:
It would also be good to add some logging output here, so we have some records of what is happening.

```javascript
const functions = require("firebase-functions");

// ..snip..

functions.logger.log("Foo:", bar);
```

At a minimum we would want something that mentions the userId and explains that they are being made an admin. Possibly also log with the email address to make it easier to search for?

Reply:
Added adjustments in 0774226691b386bb6d2e7e37fa6dc140838def6b

Reply:
I don't see the logging? https://github.com/sparkletown/sparkle/pull/1829#discussion_r668753678

Reply:
Nor did you follow this one: https://github.com/sparkletown/sparkle/pull/1829#discussion_r668750203

Reply:
moved functions 3b1e532e09e585f60dd5b0c919666db539bb8f73 and added logging 3c7148def4b25bdfe8609c4a4cec9b3b3ca91019

Reply:
added email to log output c005a930ce964aed5763d95dd152e05ba3859833